cmd/note_common.go

```diff
@@ -164,44 +228,15 @@ func createCommitComments(project string, mrID int, commit string, body string,
 			} else {
 				oldfile = f[4]
 			}
-		} else if strings.HasPrefix(scanner.Text(), "|") {
-			if comments != "" {
-				createCommitNote(project, mrID, commit, newfile, oldfile, oldLineNum, newLineNum, comments, block)
-				comments = ""
-			}
-			// read line numbers
-			fs := strings.Split(scanner.Text(), " ")
-
-			oldLineNum = -1
-			newLineNum = -1
-			for _, f := range fs {
-				if f == "" || f == "|" || f == "@@" {
-					continue
-				}
-				val, err := strconv.Atoi(f)
-				if err != nil {
-					// NaN
-					if strings.HasPrefix(f, "+") {
-						newLineNum = oldLineNum
-						oldLineNum = -1
-					}
-					break
-				} else {
-					// Number
-					if oldLineNum == -1 {
-						oldLineNum = val
-					} else {
-						newLineNum = val
-						break
-					}
-				}
-			}
-		} else {
-			// this is a comment (combine for a filename)
-			comments = comments + "\n" + scanner.Text()
+			continue
 		}
-	}
 
+		// The first character of text is preceded by a tab.  This is used in other
+		// utilities to differentiate between the diff metadata and codetext from the diff
+		// itself.
+		tabline := strings.Split(line, "\t")
```

Comment:
I didn't get the idea of this `\t` preceding the first char. Would you mind explaining it a bit more? (not sure if I missed something in the previous discussions).

Reply:
Fixed, and added a better algorithm to detect where the metadata is.  I now am using the location of the first "@" symbol in the text.  This always lines up with the metadata in the GitLab diff.  For example,
```

|          @@ -98,6 +98,8 @@ else
|  98   98    Q = @
|  99   99  endif
| 100  100  
|      101 +#somecodehere
|      102 +
| 101  103  # If the user is running make -s (silent mode), suppress echoing of
| 102  104  # commands
| 103  105  
```
so the location of the first "@" can be used to truncate the lines to find the metadata.

Reply:
Ahh , I see it now.
Thanks @prarit!